### DISCUSSION TITLE
ExLlama vs ExLlama_HF?

### DISCUSSION THREAD

**Discussion by nikshepsvn** [oobabooga/text-generation-webui#2886] [Q&A] [answered]
Currently running ExLlama and was wondering what the difference between the regular and HF versions are, anyone have any benchmarks or information?

**Comment by nikshepsvn** [comment]
@turboderp do you know?

**Comment by turboderp** [comment, marked as answer]
ExLlama_HF uses the logits from ExLlama but replaces ExLlama's sampler with the same HF pipeline used by other implementations, so that sampling parameters are interpreted the same and more samplers are supported.

**Reply by bartowski1182** [reply]
can you elaborate on any of this or provide further reading? would love to learn more, especially why it seems exllama_hf has lower VRAM use (and sure, performance) than exllama

sorry to revive old thread but not sure where else to ask
